Stackoverflow error when upgrading to v3.5.2

I tried to upgrade from Mockito v3.4.6 to v3.5.2 and I'm getting a stackoverflow error (same error with v3.5.0) when running some tests (not all of them, just a few ; even in the same class : some tests pass and others don't).

Stacktrace :
```
java.lang.StackOverflowError
	at java.base/java.lang.ThreadLocal.get(ThreadLocal.java:163)
	at org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker.lambda$new$3(InlineByteBuddyMockMaker.java:260)
	at org.mockito.internal.creation.bytebuddy.MockMethodAdvice.handleConstruction(MockMethodAdvice.java:176)
	at org.mockito.internal.creation.bytebuddy.inject.MockMethodDispatcher.handleConstruction(MockMethodDispatcher.java:53)
	at java.base/java.lang.Throwable.<init>(Throwable.java)
	at java.base/java.lang.Exception.<init>(Exception.java:66)
	at java.base/java.lang.RuntimeException.<init>(RuntimeException.java:62)
	at java.base/java.lang.ClassCastException.<init>(ClassCastException.java:58)
	at java.base/java.lang.Throwable.<init>(Throwable.java)
	at java.base/java.lang.Exception.<init>(Exception.java:66)
	at java.base/java.lang.RuntimeException.<init>(RuntimeException.java:62)
	at java.base/java.lang.ClassCastException.<init>(ClassCastException.java:58)
	at java.base/java.lang.Throwable.<init>(Throwable.java)
	at java.base/java.lang.Exception.<init>(Exception.java:66)
        ...
```

I didn't succeed in reproduce the error in a small project, but I'll spend more time to try and post the sources here when/if I have a reproducible case.

Environment :
- Mockito v3.5.2
- JUnit v5.6.2
- Java :
> openjdk version "11.0.8" 2020-07-14
> OpenJDK Runtime Environment (build 11.0.8+10-post-Ubuntu-0ubuntu118.04.1)
> OpenJDK 64-Bit Server VM (build 11.0.8+10-post-Ubuntu-0ubuntu118.04.1, mixed mode, sharing)
- Maven : v3.6.0 (tests are run by the maven-surefire plugin)

I just discovered that the error happens when I launch all my unit tests in the Maven module, but it's fine if I run the test or the test class only.

```sh
$ cd myproject/my-module-1
$ mvn test
#
# [ERROR]   MyBeanTest.test1 » StackOverflow
#
$ mvn -Dtest=MyBeanTest test
# 
# Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
#
$ mvn -Dtest=MyBeanTest#test1 test
# 
# Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
#
```

--- a/src/main/java/org/mockito/internal/creation/bytebuddy/InlineBytecodeGenerator.java
+++ b/src/main/java/org/mockito/internal/creation/bytebuddy/InlineBytecodeGenerator.java
@@ -206,6 +206,7 @@ public class InlineBytecodeGenerator implements BytecodeGenerator, ClassFileTran
         Set<Class<?>> types = new HashSet<>();
         types.add(features.mockedType);
         types.addAll(features.interfaces);
+
         synchronized (this) {
             triggerRetransformation(types, false);
         }
@@ -223,17 +224,26 @@ public class InlineBytecodeGenerator implements BytecodeGenerator, ClassFileTran
         triggerRetransformation(Collections.singleton(type), false);
     }
 
+    private static void assureInitialization(Class<?> type) {
+        try {
+            Class.forName(type.getName(), true, type.getClassLoader());
+        } catch (Throwable ignore) {
+        }
+    }
+
     private <T> void triggerRetransformation(Set<Class<?>> types, boolean flat) {
         Set<Class<?>> targets = new HashSet<Class<?>>();
 
         for (Class<?> type : types) {
             if (flat) {
                 if (!mocked.contains(type) && flatMocked.add(type)) {
+                    assureInitialization(type);
                     targets.add(type);
                 }
             } else {
                 do {
                     if (mocked.add(type)) {
+                        assureInitialization(type);
                         if (!flatMocked.remove(type)) {
                             targets.add(type);
                         }

--- a/subprojects/inline/src/test/java/org/mockitoinline/InitializationTest.java
+++ b/subprojects/inline/src/test/java/org/mockitoinline/InitializationTest.java
@@ -1,0 +1,25 @@
+/*
+ * Copyright (c) 2020 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockitoinline;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static junit.framework.TestCase.assertEquals;
+
+public class InitializationTest {
+
+    @Test
+    public void assure_initialization_prior_to_instrumentation() {
+        @SuppressWarnings("unused")
+        SampleEnum mock = Mockito.mock(SampleEnum.class);
+        SampleEnum[] values = SampleEnum.values();
+        assertEquals("VALUE", values[0].name());
+    }
+
+    public enum SampleEnum {
+        VALUE
+    }
+}

--- a/subprojects/inline/src/test/java/org/mockitoinline/PluginTest.java
+++ b/subprojects/inline/src/test/java/org/mockitoinline/PluginTest.java
@@ -7,14 +7,20 @@ package org.mockitoinline;
 import org.junit.Test;
 import org.mockito.internal.configuration.plugins.Plugins;
 import org.mockito.internal.creation.bytebuddy.InlineByteBuddyMockMaker;
+import org.mockito.internal.util.reflection.ModuleMemberAccessor;
 
 import static org.junit.Assert.*;
 
 public class PluginTest {
 
     @Test
-    public void plugin_type_should_be_inline() throws Exception {
+    public void mock_maker_should_be_inline() throws Exception {
         assertTrue(Plugins.getMockMaker() instanceof InlineByteBuddyMockMaker);
+    }
+
+    @Test
+    public void member_accessor_should_be_module() throws Exception {
+        assertTrue(Plugins.getMemberAccessor() instanceof ModuleMemberAccessor);
     }
 
 }

